json_response.go

```diff
@@ -110,3 +110,18 @@ func ReadResponse(r io.Reader, v interface{}) error {
 	_, err := ReadResponsePage(r, v)
 	return err
 }
+
+// ReadError attempts to unmarshal JSON-encoded error details from the supplied reader. It returns
+// nil if an error could not be parsed from the response, or if the parsed error was nil.
+func ReadError(r io.Reader) error {
+	var u struct {
+		Error *Error `json:"error"`
+	}
+	if err := json.NewDecoder(r).Decode(&u); err != nil {
```

Comment:
Is it a problem that we're losing the rest of the data in `r` here?

Reply:
Not sure what you mean, I don't think there is a way to deterministically do a partial read using the `json` package? And not sure what the use case would be if there were?